java/ql/src/Violations of Best Practice/Magic Constants/MagicConstants.qll

```diff
@@ -296,14 +293,7 @@ private predicate firstOccurrence(Literal lit, string value, string context, int
   )
 }
 
-predicate isNumber(Literal lit) {
-  lit.getType().getName() = "char" or
-  lit.getType().getName() = "short" or
-  lit.getType().getName() = "int" or
-  lit.getType().getName() = "long" or
-  lit.getType().getName() = "float" or
-  lit.getType().getName() = "double"
-}
+predicate isNumber(Literal lit) { lit.getType() instanceof NumericOrCharType }
```

Comment:
Changes behaviour by adding `byte` and boxed types

Reply:
That is what I mentioned in the description. Sorry, would probably have been better to add a comment instead.
Was it intentional that `byte` was excluded? (I can also revert the change though)

Boxed types should not be an issue (except for performance?) because literals are always primitive, except for String but that is not a numeric type.

Reply:
It's fine to include `byte`. The dependent queries are low-precision recommendation queries so it doesn't matter too much.